### PR TITLE
Track site_created event

### LINKED ISSUE #9959: Analytics: Resume bumping wpandroid_site_created
For more context see the comment thread at p77Llu-c4u-p2
With the introduction of the new site creation flow back in March, we stopped bumping the `wpandroid_site_created` event in favor of `wpandroid_enhanced_site_creation_completed`.  

To address some confusion in the event names and other dashboards that might have been impacted, we'd like to resume bumping the `wpandroid_site_created` event.  We want to stop bumping `wpandroid_enhanced_site_creation_completed` since it will make it easier to aggregate data across both events.

## Matching iOS Issue:

https://github.com/wordpress-mobile/WordPress-iOS/issues/11809

### PR DESCRIPTION
**Pull request by malinajirka**
~`Not-ready-for-review` -> I'd like to wait for an input from @aerych as I haven't been following the related discussions. Eric, if you agree with the proposed solution, please remove the "Not Ready for Review" label. If you have some comments, let me know and I'll make any necessary changes;). Thanks!~

Fixes #9959 

I've created two new events
1.  ENHANCED_SITE_CREATION_PREVIEW_OK_BUTTON_TAPPED - emitted when the user presses "OK" button on the Site Preview screen (previously called "ENHANCED_SITE_CREATION_COMPLETED")

2. SITE_CREATED - emitted when we believe the site has been successfully created
    a. We've received a successful response from the server
    b. We've received `SITE_NAME_EXISTS` error, but we are in retry mode -> we assume the site was already created, but the response from the server got lost. This assumption might be wrong in extreme edge cases, but we decided to live with it as implementing bulletproof solution would be too much effort for too little gain.

Note 2b. The "site_created" event wasn't being recorded in this case in the old site creation flow. However, I believe we want to record it to make sure the number of recorded events is as close to the number of created sites as possible.

To test:
My -> App Settings -> Privacy Settings -> Check Collect information switch

1. Start the Site Creation flow
2. Open LogCat
3. Finish the flow and notice LogCat contains `Tracked: site_created` msg
4. Click on the ok button on the SitePreview screen and notice LogCat contains `Tracked: enhanced_site_creation_preview_ok_button_tapped`
----------
1. Start the Site Creation flow
2. Open LogCat
3. Turn on airplane mode right after you pick a domain and click on the "Create Site" button
4. Notice an error screen is shown
5. Turn off airplane mode
6. Click on Retry  and notice LogCat contains `Tracked: site_created` msg
-----------
1. Start the Site Creation flow
2. Open LogCat
3. Press home button (put the app into background) right after you pick a domain and click on the "Create Site" button
4. Notice LogCat contains `Tracked: site_created` msg

@aerych Can you please confirm this all makes sense. Thanks

cc @diegoreymendez As we'll probably want to preserve cross-platform parity.

Update release notes:

- There are no user facing changes.

### CODE REVIEW

**Comment by aerych** [comment]
Hey @malinajirka.  It all sounds good to me!  Thanks for thinking of the retry scenario and ensuring site_created is bumped then. :)   I'm also a fan of the introduction of `ENHANCED_SITE_CREATION_PREVIEW_OK_BUTTON_TAPPED` taking place of the enhanced site created event.

**Comment by diegoreymendez** [comment]
I'm not sure about point 2b.  I understand why we're trying to account for a possible failure in the first request.

But doesn't this also mean we can get "invalid" bumps as well now?  Which one is more likely to happen?

**Comment by malinajirka** [comment]
>This assumption might be wrong in extreme edge cases
This is the **extremely** rare case when the assumptions isn't true.
1. User A request list of domains 
2. User B request list of domains before user A picked a domain
3. User A picks domain "test.wordpress.com" and click on create site button
4. User B picks domain "test.wordpress.com" and click on create site button 
5. The site creation request for user A timeouts.
6. User B receives "SITE_NAME_EXISTS" response, but since they weren't retrying, the SiteCreation fails without bumping any track events.
7. User A clicks on retry -> server replies with "SITE_NAME_EXISTS", but since user A is retrying, the "site_created" event is sent and a preview screen with a newly created site is presented to the user.

>But doesn't this also mean we can get "invalid" bumps as well now? 

Since we bump the event **after we receive** either a successful response from the server or error response ("SITE_NAME_EXISTS") response from the server + we are retrying, I believe we will never have "invalid" bumps.

**Comment by diegoreymendez** [comment]
The bit of info I was not considering (that confused me to begin with) is that we don't allow the user to just pick any domain, but only a domain that's available to begin with.

Your reply coupled with this makes a lot of sense.  Thanks @malinajirka!

File: WordPress/src/main/java/org/wordpress/android/ui/sitecreation/misc/SiteCreationTracker.kt [hunk 1]
```diff
@@ -93,8 +93,12 @@ class SiteCreationTracker @Inject constructor(val tracker: AnalyticsTrackerWrapp
         tracker.track(AnalyticsTracker.Stat.ENHANCED_SITE_CREATION_SUCCESS_PREVIEW_LOADED)
     }
 
-    fun trackCreationCompleted() {
-        tracker.track(AnalyticsTracker.Stat.ENHANCED_SITE_CREATION_COMPLETED)
+    fun trackPreviewOkButtonTapped() {
+        tracker.track(AnalyticsTracker.Stat.ENHANCED_SITE_CREATION_PREVIEW_OK_BUTTON_TAPPED)
```
Comment (maxme) [inline WordPress/src/main/java/org/wordpress/android/ui/sitecreation/misc/SiteCreationTracker.kt]: Should we update the document tracking our event names?

Comment (maxme) [reply WordPress/src/main/java/org/wordpress/android/ui/sitecreation/misc/SiteCreationTracker.kt]: And open a ticket on the iOS side to rename [this](https://github.com/wordpress-mobile/WordPress-iOS/search?q=ENHANCED_SITE_CREATION_COMPLETED&unscoped_q=ENHANCED_SITE_CREATION_COMPLETED)

Comment (diegoreymendez) [reply WordPress/src/main/java/org/wordpress/android/ui/sitecreation/misc/SiteCreationTracker.kt]: iOS Issue: https://github.com/wordpress-mobile/WordPress-iOS/issues/11809
iOS PR: https://github.com/wordpress-mobile/WordPress-iOS/pull/11829

Comment (malinajirka) [reply WordPress/src/main/java/org/wordpress/android/ui/sitecreation/misc/SiteCreationTracker.kt]: >Should we update the document tracking our event names?

Yes, I always update the document after the PR is merged. However, it makes sense to do it before the PR so the reviewer can review it as well! 

Updated!

Comment (malinajirka) [reply WordPress/src/main/java/org/wordpress/android/ui/sitecreation/misc/SiteCreationTracker.kt]: >And open a ticket on the iOS side to rename this

I thought updating it myself would be confusing as Diego wouldn't have the background and might not even notice the description changed so I just pinged him. Or do you think it's better to update it directly?

Thanks Diego for updating the iOS repo ;)
